### PR TITLE
feature(kumactl) added logging and tracing information

### LINKED ISSUE #628: Add logging information to kumactl get meshes
With a list of backend names configured for logging. If no logging is specified, then return `off`.

```sh
./kumactl get meshes
NAME      mTLS   CA        METRICS  LOGGING                                        TRACING
default   on     builtin   prometheus  logging-1, logging-2, logging-3   backend-1, backend-2
```

### PR DESCRIPTION
**Pull request by tharun208**
Feature #628

### Summary
Added Logging and Tracing Information for meshes
### Full changelog

*Implemented logging and tracing information for meshes
* added helper functions for getting logging and tracing information and tests
* updated tests

### Issues resolved

Fix #628

### CODE REVIEW

File: pkg/core/resources/apis/mesh/mesh_helpers.go [hunk 1]
```diff
@@ -36,3 +38,29 @@ func (m *MeshResource) GetTracingBackend(name string) *mesh_proto.TracingBackend
...
+// if empty return default logging
+func (m *MeshResource) GetLoggingBackends() string {
+	var backends []string
+	if len(m.Spec.GetLogging().GetBackends()) != 0 {
+		for _, backend := range m.Spec.GetLogging().GetBackends() {
+			backends = append(backends, backend.GetName())
+		}
+		return strings.Join(backends, ", ")
+	}
+	return m.Spec.GetLogging().GetDefaultBackend()
```
Comment (jakubdyszkiewicz) [inline pkg/core/resources/apis/mesh/mesh_helpers.go:52]: default backend is one of the existing backend. There is no situation that there are 0 backends and there is default backend. I think the logic should be just
```
var backends []string
for _, backend := range m.Spec.GetLogging().GetBackends() {
	backends = append(backends, backend.GetName())
}
return strings.Join(backends, ", ")
```
(it should return `""` when there are no backends)

Comment (jakubdyszkiewicz) [reply pkg/core/resources/apis/mesh/mesh_helpers.go:52]: Same with `GetTracingBackends()`

File: app/kumactl/cmd/get/get_meshes_test.go [hunk 2]
```diff
@@ -76,12 +76,64 @@ var _ = Describe("kumactl get meshes", func() {
 						Path: "/non-standard-path",
 					},
 				},
+				Logging: &v1alpha1.Logging{
+					DefaultBackend: "default-backend",
+					Backends: []*v1alpha1.LoggingBackend{
+						{
+							Name: "backend-1",
+						},
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/get/get_meshes_test.go:84]: Can we provide more complete and valid backend here?
Aside from name we should also specify the type of the backend and required fields. Feel free to consult kuma.io as well as try to apply YAML with example via kumactl to trigger validation

File: pkg/core/resources/apis/mesh/mesh_helpers.go (hunk 1 above)
Comment (tharun208) [reply pkg/core/resources/apis/mesh/mesh_helpers.go:52]: then, if it is empty means we need to show logging as "off" right as well for tracing ?

Comment (jakubdyszkiewicz) [reply pkg/core/resources/apis/mesh/mesh_helpers.go:52]: Yes

File: pkg/core/resources/apis/mesh/mesh_helpers.go [hunk 3]
```diff
@@ -36,3 +38,29 @@ func (m *MeshResource) GetTracingBackend(name string) *mesh_proto.TracingBackend
 	}
 	return backends[name]
 }
+
+// GetLoggingBackends will return logging backends as comma separated strings
+// if empty return default logging
```
Comment (jakubdyszkiewicz) [inline pkg/core/resources/apis/mesh/mesh_helpers.go:43]: `if empty return default logging` that's no longer true

File: pkg/core/resources/apis/mesh/mesh_helpers.go [hunk 4]
```diff
@@ -36,3 +38,29 @@ func (m *MeshResource) GetTracingBackend(name string) *mesh_proto.TracingBackend
...
+func (m *MeshResource) GetLoggingBackends() string {
+	var backends []string
+	for _, backend := range m.Spec.GetLogging().GetBackends() {
+		backends = append(backends, backend.GetName())
+	}
+	if len(backends) > 0 {
```
Comment (jakubdyszkiewicz) [inline pkg/core/resources/apis/mesh/mesh_helpers.go:49]: is this if necessary? I think that `strings.Join(backends, ", ")` returns empty string when the array is empty

File: pkg/core/resources/apis/mesh/mesh_helpers.go [hunk 5]
```diff
@@ -36,3 +38,29 @@ func (m *MeshResource) GetTracingBackend(name string) *mesh_proto.TracingBackend
...
+		return strings.Join(backends, ", ")
+	}
+	return ""
+}
+
+// GetTracingBackends will return tracing backends as comma separated strings
+// if empty return default tracing
```
Comment (jakubdyszkiewicz) [inline pkg/core/resources/apis/mesh/mesh_helpers.go:56]: and here too

**Comment by tharun208** [comment]
should I need to rebase with upstream?

**Comment by jakubdyszkiewicz** [comment]
yes please

**Comment by jakubdyszkiewicz** [comment]
thanks for working on this!

**Comment by subnetmarco** [comment]
Thanks @tharun208 !

**Pull request merged**